repair/row_level.cc

```diff
@@ -1383,14 +1393,58 @@ class repair_meta {
             // and _working_row_buf are ordered.
             utils::merge_to_gently(_working_row_buf, row_diff,
                  [this] (const repair_row& x, const repair_row& y) { return _cmp(x.boundary(), y.boundary()) < 0; });
+            for (auto& r : row_diff) {
+                thread::maybe_yield();
+                _working_row_buf_combined_hash.add(r.hash());
+            }
         }
         if (update_hash_set) {
             _peer_row_hash_sets[node_idx] = boost::copy_range<repair_hash_set>(row_diff |
                     boost::adaptors::transformed([] (repair_row& r) { thread::maybe_yield(); return r.hash(); }));
         }
-        do_apply_rows(std::move(row_diff), node_idx, update_buf).get();
+        // Repair rows in row_diff will be flushed to disk by flush_rows_in_working_row_buf,
+        // so we skip calling do_apply_rows here.
+        _flush_to_disk = true;
+    }
+public:
+    // Must run inside a seastar thread
+    void flush_rows_in_working_row_buf() {
+        if (_flush_to_disk) {
+            _flush_to_disk = false;
+        } else {
+            return;
+        }
+        auto cmp = position_in_partition::tri_compare(*_schema);
+        lw_shared_ptr<mutation_fragment> last_mf;
+        lw_shared_ptr<const decorated_key_with_hash> last_dk;
+        for (auto& r : _working_row_buf) {
+            thread::maybe_yield();
+            if (!r.flush_to_disk()) {
+                continue;
+            }
+            _repair_writer->create_writer(_db, node_idx);
+            auto mf = r.get_mutation_fragment_ptr();
+            const auto& pk = r.get_dk_with_hash()->dk.key();
+            if (last_mf &&
+                last_dk &&
+                cmp(last_mf->position(), mf->position()) == -1 &&
+                pk.legacy_equal(*_schema, last_dk->dk.key()) &&
+                last_mf->mergeable_with(*mf)) {
```

Comment:
Please indent these lines with double indtent, really hard to see where the if header ends and where the body starts.

Reply:
I prefer to align to first the item in if because it looks nicer. I agree double indent is easier to read though.